File: benchmarks/executable/restaurant_uint_bit_not.c
// C23 reference for the Restaurant UInt bitwise-complement workload.

#include <inttypes.h>
#include <stdint.h>
#include <stdio.h>

#ifdef _WIN32
#include <fcntl.h>
#include <io.h>
#endif

static volatile uint64_t runtime_zero = UINT64_C(0);

int main(void) {
#ifdef _WIN32
    if (_setmode(_fileno(stdout), _O_BINARY) == -1) return 1;
#endif
    const uint64_t inverted = ~runtime_zero;
    return printf("UInt not %" PRIu64 "\n", inverted) < 0 ? 1 : 0;
}

File: benchmarks/executable/restaurant_uint_bit_not.rs
// Rust 2024 reference for the Restaurant UInt bitwise-complement workload.

use std::hint::black_box;
use std::io::{self, Write};

fn main() {
    let inverted = !black_box(0_u64);
    let mut stdout = io::stdout().lock();
    if writeln!(stdout, "UInt not {inverted}").is_err() {
        std::process::exit(1);
    }
}
